packages/devtools_app/lib/src/charts/chart_controller.dart

```diff
@@ -556,4 +565,53 @@ class ChartController extends DisposableController
     }
     return -1;
   }
+
+  int xCoordToTimestamp(double xCoord) =>
+      timestamps[xCoordToTimestampIndex(xCoord)];
+
+  int xCoordToTimestampIndex(double xCoord) {
+    final firstVisibleIndex = leftVisibleIndex;
+    final index = (xCoord - leftPadding) ~/ tickWidth;
+
+    int timestampedIndex;
+
+    if (totalTimestampTicks < 0) {
+      // Not enough items plotted on x-axis.
+      timestampedIndex = index + totalTimestampTicks;
+    } else if (index >= firstVisibleIndex) {
+      timestampedIndex = index - firstVisibleIndex;
+    } else {
+      timestampedIndex = index + firstVisibleIndex;
+    }
+
+    // If index is left side, negative, that has yet to have any data
+    // return the first timestamp.
+    return timestampedIndex >= 0 ? timestampedIndex : 0;
+  }
+}
+
+/// When tap occurs in a chart compute the timestamp and index clicked.
```

Comment:
TapNotifier seems like a confusing name as this doesn't do any notifying.
Seems more like a TapLocation.

Word compute also seems wrong as the timestamp is passed in not computed.